File: java/src/main/java/S025.java
import com.google.common.math.BigIntegerMath;

import java.math.BigInteger;
import java.math.RoundingMode;

public class S025 {

    private static final int DIGITS = 1000;

    public static int solve() {
        BigInteger left = BigInteger.ONE;
        BigInteger right = BigInteger.ONE;
        for (int i = 3;; i++) {
            BigInteger next = left.add(right);
            int digits = BigIntegerMath.log10(next, RoundingMode.DOWN) + 1;
            if (digits == DIGITS) {
                return i;
            }
            left = right;
            right = next;
        }
    }

}

File: cpp/S025.cpp
#include <iostream>
#include <functional>
#include "maths.cpp"

using namespace std;

static const int TARGET = 1000;

BigInteger get_target() {
    BigInteger target = BigInteger::ONE();
    for (int i = 1; i < TARGET; i++) {
        target = target.times(10);
    }
    return target;
}

int main() {
    BigInteger target = get_target();
    BigInteger left = BigInteger::ONE();
    BigInteger right = BigInteger::ONE();
    for (int i = 3;; i++) {
        BigInteger next = left.add(right);
        if (next.compare(target) >= 0) {
            cout << i << endl;
            return 0;
        }
        left = right;
        right = next;
    }
}
